### DISCUSSION TITLE
Option to rename only applicationId while keeping package name.

### DISCUSSION THREAD

**Discussion by MikuAuahDark** [iBotPeaches/Apktool#3163] [Ideas]
**Is your feature request related to a problem? Please describe.**
In newer Android, "package name" and "applicationId" are no longer interchangeable. However in APKTool 2.7.0, changing the `package` in AndroidManifest.xml (or specifying `renameManifestPackage` in `apktool.yml`) changes both, and without proper checking through the smali files, will result in app fails to launch because it looks for wrong `R` classes.

**Describe the solution you'd like**
Option in `apktool.yml` to specify the new `applicationId` or command-line option to override the `applicationId`. This will keep "package name" intact but changes the `applicationId` alone.

**Additional context**
https://developer.android.com/studio/build/configure-app-module

> **Note**: The application ID used to be directly tied to your code's package name, so some Android APIs use the term "package name" in their method names and parameter names. This is actually your application ID. For example, the [`Context.getPackageName()`](https://developer.android.com/reference/android/content/Context#getPackageName()) method **returns your application ID**. There's never a need to share your code's true package name outside your app code.

Quick look at AAPT2, it seems possible with `--rename-resources-package`.

**Comment by iBotPeaches** [comment]
Seems like a sane request. Added label.
